lib/prometheus_exporter/middleware.rb

```diff
@@ -6,12 +6,38 @@
 class PrometheusExporter::Middleware
   MethodProfiler = PrometheusExporter::Instrumentation::MethodProfiler
 
+  module RedisInstrumenter
+    def call(_command, _config)
+      unless prof = Thread.current[:_method_profiler]
+        super
+      end
+
+      start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      super
+    ensure
+      data = (prof[:redis] ||= { duration: 0.0, calls: 0 })
+      data[:duration] += Process.clock_gettime(Process::CLOCK_MONOTONIC) - start
+      data[:calls] += 1
```

Comment:
Will we be able to improve the API for `MethodProfiler` such that we don't have to re implement parts of what `MethodProfiler` does?

Reply:
tricky cause of performance sensitivity, we will eat an extra method call for instrumentation :cry: not sure I love that

Reply:
Technically, switching to the middleware for instrumentation is already adding an extra method call. I do wonder if an extra method call has any significance on performance in the big scheme of things. Even the profiling code itself is already making multiple additional method calls and I'm not convinced an extra method call will move the needle much.